packages/dds/shared-object-base/src/types.ts

```diff
@@ -8,6 +8,18 @@ import { IChannel } from "@fluidframework/datastore-definitions";
 import { ISequencedDocumentMessage } from "@fluidframework/protocol-definitions";
 import { IGarbageCollectionData } from "@fluidframework/runtime-definitions";
 
+/**
+ * Events emitted by {@link ISharedObject}.
+ *
+ * @remarks
+ * ### "pre-op"
```

Comment:
The "pre-op" event probably should be removed, or at minimum we should discourage its use.  It fires just before an incoming op is applied to the shared object.